addons/onboarding/controllers/onboarding.py

```diff
@@ -12,11 +13,22 @@ def get_onboarding_data(self, route_name=None):
             return {}
 
         onboarding = request.env['onboarding.onboarding'].search([('route_name', '=', route_name)])
-        if onboarding and not onboarding._search_or_create_progress().is_onboarding_closed:
+        if onboarding and not self._safe_search_or_create_progress(onboarding).is_onboarding_closed:
             # JS implementation of the onboarding panel expects this data structure
             return {
                 'html': request.env['ir.qweb']._render(
                     'onboarding.onboarding_panel', onboarding._prepare_rendering_values())
             }
 
         return {}
+
+    @staticmethod
+    def _safe_search_or_create_progress(onboarding):
+        try:
+            current_progress = onboarding._search_or_create_progress()
+        except ValidationError as e:
+            # If initially not found but was created by another process before we tried to create
+            current_progress = onboarding.current_progress_id
```

Comment:
Hello @Julien00859 ,
We discussed this with @xmo-odoo (https://github.com/odoo/odoo/pull/104223#discussion_r1068009272) (larger PR that includes this commit).
Do you know a way to bypass `REPEATABLE_READ` so we can actually look again in the database if a record has been created since our previous lookup?

Reply:
I don't know any and I don't think that would be a good solution :/